File: except_self/except_self.js
function exceptSelf(nums) {
  let answer = [1]

  for(i = 1; i < nums.length;i++){
    answer[i] = answer[i-1] * nums[i-1]
  }

  let r = 1
  for(j = (nums.length-1); j > -1; j--){
    answer[j] *= r
    r *= nums[j]
  }
  return answer
}


console.log(exceptSelf([1,2,3,4]))

File: except_self/except_self.rb
def product_except_self(nums)
  answer = [1]

  (1...nums.length).each do |i|
    answer[i] = answer[i-1] * nums[i-1]
  end

  r = 1
  (nums.length-1).downto(0) do |i|
    answer[i] *= r
    r *= nums[i]
  end

  answer

end


p product_except_self([1,2,3,4]) #[24,12,8,6]
